### DISCUSSION TITLE
Test edge cases

### DISCUSSION THREAD

**Discussion by maurolepore** [2DegreesInvesting/tiltIndicator#247] [General]
.

**Comment by maurolepore** [comment]
@AnneSchoenauer you said you already thought of a few edge-cases. Can you list them here please?

**Comment by AnneSchoenauer** [comment]
Dear @maurolepore and @Tilmon, 
the edge case that I thought about is with regard to the benchmarks. As I am a bit confused if benchmarks are already implemented or if it another feature - I leave it to you @Tilmon to priotise this edge case. When we don't implement the benchmarks yet, I think it might not be so urgent. If we do then I think the edge case is quite interesting. 
To re-cap we have six different benchmarks - all, all_unit, sector, sector_unit, isic, isic_unit. The differences between those six are that one product is compared to all products (all), to all products with the same unit (all_unit) and so on. You can see this in this [xls sheet](https://docs.google.com/spreadsheets/d/1wIIqJdGAD3Tg3URQ2oDTxL4UvI-SEd5VdwJiCSW1JyM/edit#gid=0): 
I now see two problems: 
1. The more specific the comparison group is (isic_unit is the most specific one), the more likely it is that there are not enough products in one category to compare them to each other. As we say all products that are in the lower 33% have a low risk. Every product between 33-66% is medium and every product that is in the higher percentile (all over 66%) we need at least three products to compare the products to each other. 
2. What is if we cannot divide the amount of products by 3? How do we make the cut between what is in the lower, medium and higher category? Let's say we have five products - where do we make the cut of what is low, medium and high? 
Both problems become clearer when you look in the xls sheet I think!
Let me know if this makes sense - also happy to try to explain it in greater detail!
All the best
Anne

**Comment by Tilmon** [comment]
@AnneSchoenauer @maurolepore I think that's a valid point. I think we can test whether these edge cases exist by looking at the ecoinvent data and checking whether each benchmark combination has >3 products. Latest ecoinvent data can be found [here](https://docs.google.com/spreadsheets/d/1AYfiS7CpLy8QjuZTsHEeLOKdj9tgVznN/edit?usp=share_link&ouid=110874146095719873200&rtpof=true&sd=true). 
If it turns out that these edge cases exist, I suggest the following: 

1. If we have < 3 products per benchmark combination, I suggest to put `NA`s as shares for the respective benchmark. 
2. Good question, but I would assume that the code can still calculate percentiles even if we have e.g. 5 products. 5/3 = 1.6666. This would mean that the first 2 products are low, the 1 in the middle is medium, and 2 are high. This assumes the code rounds up, which would need to be double checked. 

What do you think @AnneSchoenauer and @maurolepore ?
